src/datadog/grok/grok_filter.rs

```diff
@@ -195,6 +213,54 @@ pub fn apply_filter(value: &Value, filter: &GrokFilter) -> Result<Value, GrokRun
                 value.to_string(),
             )),
         },
+        GrokFilter::Rubyhash => match value {
+            Value::Bytes(bytes) => parse_ruby_hash(String::from_utf8_lossy(&bytes).as_ref())
+                .map_err(|_e| {
+                    GrokRuntimeError::FailedToApplyFilter(filter.to_string(), value.to_string())
+                }),
+            _ => Err(GrokRuntimeError::FailedToApplyFilter(
+                filter.to_string(),
+                value.to_string(),
+            )),
+        },
```

Comment:
Could you reduce this repeated pattern to a common function that handles the `match value`, `String` conversion, and error return?

Reply:
good idea, refactored it.